xtable-core/src/main/java/org/apache/xtable/hudi/HudiConversionTarget.java

```diff
@@ -316,6 +323,53 @@ public String getTableFormat() {
     return TableFormat.HUDI;
   }
 
+  @Override
+  public Optional<String> getTargetCommitIdentifier(String sourceIdentifier) {
+    if (!metaClient.isPresent()) {
+      return Optional.empty();
+    }
+    return getTargetCommitIdentifier(sourceIdentifier, metaClient.get());
+  }
+
+  public Optional<String> getTargetCommitIdentifier(
+      String sourceIdentifier, HoodieTableMetaClient metaClient) {
+    long sourceIdentifierVal = Long.parseLong(sourceIdentifier);
+
+    HoodieTimeline completedTimeline = metaClient.getActiveTimeline().filterCompletedInstants();
+
+    for (HoodieInstant instant : completedTimeline.getInstants()) {
+      try {
+        Option<byte[]> instantDetails = metaClient.getActiveTimeline().getInstantDetails(instant);
```

Comment:
There is a similar block of code above (line 300) that also handles parsing of the "replace commit" type which is what we'll be writing from XTable. Can you extract that into a common helper method to avoid repeated code?